rpc/server_auth.go

```diff
@@ -210,14 +217,22 @@ func (ss *simpleServer) ensureAuthed(ctx context.Context) (interface{}, error) {
 			}
 		}
 		if verifiedCert == nil {
-			return nil, err
+			return "", nil, err
 		}
 		if tlsAuthEntity, tlsErr := ss.tlsAuthHandler(ctx, verifiedCert.DNSNames...); tlsErr == nil {
-			return tlsAuthEntity, nil
+			// mTLS based authentication contexts do not really have a sense of a unique identifier
+			// when considering multiple clients using the certificate. We deem this okay but it does
+			// mean that if the identifier is used to bind to the concept of a unique session, it is
+			// not sufficient without another piece of information (like an address and port).
+			// Furthermore, if TLS certificate verification is disabled, this trust is lost.
+			// Our best chance at uniqueness with a compliant CA is to use the issuer DN (Distinguished Name)
+			// along with the serial number; compliancy hinges on issuing unique serial numbers and if this
+			// is an intermediate CA, their parent issuing unique DNs.
+			return verifiedCert.Issuer.String() + ":" + verifiedCert.SerialNumber.String(), tlsAuthEntity, nil
```

Comment:
my research tells me this is safe to use as a unique identifier. unfortunately the TLS session/unique ID is not safe and the security considerations are too complex for me to understand if they are safe in our cases, so I did not use it

Reply:
Did you look at Thumbprint as an option?

Reply:
I did but I opted for this as actual uniqueness over a thumbprint which is still less unique. Do you have a preference?

Reply:
I don't have a strong preference. My understanding was that thumbprint will at least always be there and outside of sha-1 collisions should be quite unique.